### DISCUSSION TITLE
Typescript useController error control type

### DISCUSSION THREAD

**Discussion by Paolo-Gambardella** [react-hook-form/react-hook-form#6343] [Q&A]
I need help for typing my useController components. I guess my problem is that my actual typing is not enough generic.

Here is my use case:

`<TextInputForm name="phone" placeholder="Phone" control={control} />`

My Component `TextInputForm` is typed like that

`export type TextInputFormProps = Pick<UseControllerProps, 'control' | 'name' | 'rules'> &
  TextInputProps`

`TextInputProps` is just a bunch of personal props (nothing important here)

Actually it's working perfectly but i have a typescript error that i would like to fix

`Type 'Control<{ firstname: string; phone: string; password: string; }, object>' is not assignable to type 'Control<FieldValues, object>'.
  Types of property '_updateProps' are incompatible.
    Type '(props: Partial<{ mode: keyof ValidationMode; reValidateMode: "onBlur" | "onChange" | "onSubmit"; defaultValues: { firstname?: string | undefined; phone?: string | undefined; password?: string | undefined; }; ... 6 more ...; delayError: number; }>) => void' is not assignable to type '(props: Partial<{ mode: keyof ValidationMode; reValidateMode: "onBlur" | "onChange" | "onSubmit"; defaultValues: { [x: string]: any; }; resolver: Resolver<FieldValues, object>; ... 5 more ...; delayError: number; }>) => void'.
      Types of parameters 'props' and 'props' are incompatible.
        Type 'Partial<{ mode: keyof ValidationMode; reValidateMode: "onBlur" | "onChange" | "onSubmit"; defaultValues: { [x: string]: any; }; resolver: Resolver<FieldValues, object>; ... 5 more ...; delayError: number; }>' is not assignable to type 'Partial<{ mode: keyof ValidationMode; reValidateMode: "onBlur" | "onChange" | "onSubmit"; defaultValues: { firstname?: string | undefined; phone?: string | undefined; password?: string | undefined; }; ... 6 more ...; delayError: number; }>'.
          Types of property 'resolver' are incompatible.
            Type 'Resolver<FieldValues, object> | undefined' is not assignable to type 'Resolver<{ firstname: string; phone: string; password: string; }, object> | undefined'.
              Type 'Resolver<FieldValues, object>' is not assignable to type 'Resolver<{ firstname: string; phone: string; password: string; }, object>'.ts(2322)
controller.d.ts(22, 5): The expected type comes from property 'control' which is declared here on type 'IntrinsicAttributes & Pick<UseControllerProps<FieldValues, string>, "control" | "name" | "rules"> & Omit<TextInputProps, "value" | "onChangeText"> & { ...; }'`

Does someone have the solution to my problem ? 

Thanks guys !

**Comment by Paolo-Gambardella** [comment]
@bluebill1049 save me please 😭

**Reply by bluebill1049** [reply]
`const TextInputForm = ({ control }: {Control<FormVlaues}) => <input />` did you pass the generic to the `Control` type? and provide a codesandbox.
